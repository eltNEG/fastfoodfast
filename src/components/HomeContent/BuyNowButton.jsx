import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

const BuyNowButton = ({ foodId }) => (
  <Link to={`/checkout/${foodId}`}>
    <button>Buy Now</button>
  </Link>
);

BuyNowButton.defaultProps = {
  foodId: '-1'
};

BuyNowButton.propTypes = {
  foodId: PropTypes.string
};

export default BuyNowButton;
