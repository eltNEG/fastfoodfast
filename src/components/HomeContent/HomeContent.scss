@import '../../styles/styles';

.home-content {
    @extend %content;
    width: 100%;
    .food-list {
        display: flex;
        align-self: center;
        justify-self: center;
        button {
            @extend %fff-btn;
            height: 2.5rem;
            width: 45%;
            min-width: 8rem;
            background-color: $dark-blue;
            color: $white;
            border-radius: $border-radius;
            font-size: 1.2rem;
        }
    }
    .ui.card{
        .ui.image{
            width: 100%;
            height: 15rem;
        }
    }
}