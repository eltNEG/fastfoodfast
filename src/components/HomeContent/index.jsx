import React from 'react';
import PropTypes from 'prop-types';
import { Card } from 'semantic-ui-react';
import './HomeContent.scss';
import cardItems from '../../helpers';
import BuyNowBtn from './BuyNowButton';

const HomeContent = ({ foodItems }) => (
  <div className="home-content">
    <Card.Group
      stackable
      doubling
      className="food-list"
      itemsPerRow={4}
      items={cardItems(foodItems, <BuyNowBtn />)}
    />
  </div>
);

HomeContent.propTypes = {
  foodItems: PropTypes.instanceOf(Array).isRequired
};

export default HomeContent;
