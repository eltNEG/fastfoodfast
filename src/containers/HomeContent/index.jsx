import React from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import PropTypes from 'prop-types';
import HomeContent from '../../components/HomeContent';
import { doFetchFoods } from '../../actions/foodActions';

class HomeContentContainer extends React.Component {
  componentDidMount() {
    const { doFetchFoods } = this.props;
    doFetchFoods();
  }

  render() {
    const { foodItems } = this.props;
    return <HomeContent foodItems={foodItems} />;
  }
}

const mapStateToProps = (state) => {
  const { foodReducer } = state;
  const { foodItems } = foodReducer;
  return { foodItems };
};

const mapDispatchToProps = dispatch => bindActionCreators({ doFetchFoods }, dispatch);

HomeContentContainer.propTypes = {
  foodItems: PropTypes.instanceOf(Array).isRequired,
  doFetchFoods: PropTypes.func.isRequired
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(HomeContentContainer);
