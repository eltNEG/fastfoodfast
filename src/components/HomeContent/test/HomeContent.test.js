import React from 'react';
import { shallow } from 'enzyme';
import HomeContent from '..';
import BuyNowBtn from '../BuyNowButton';

describe('HomeContent snapshot', () => {
  it('should match HomeContent.jsx snapshot', () => {
    const wrapper = shallow(<HomeContent />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe('BuyNowBtn snapshot', () => {
  it('should match BuyNowBtn.jsx snapshot', () => {
    const wrapper = shallow(<BuyNowBtn />);
    expect(wrapper).toMatchSnapshot();
  });
});
