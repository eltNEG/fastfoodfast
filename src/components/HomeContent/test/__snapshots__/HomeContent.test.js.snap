// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BuyNowBtn snapshot should match BuyNowBtn.jsx snapshot 1`] = `
<button>
  Buy Now
</button>
`;

exports[`HomeContent snapshot should match HomeContent.jsx snapshot 1`] = `
<div
  className="home-content"
>
  <CardGroup
    className="food-list"
    doubling={true}
    items={Array []}
    itemsPerRow={4}
    stackable={true}
  />
</div>
`;
